kornia/color/raw.py

```diff
@@ -211,6 +211,64 @@ def rgb_to_raw(image: torch.Tensor, cfa: CFA) -> torch.Tensor:
     return output
 
 
+def raw_to_rgb_2x2_downscaled(image: torch.Tensor, cfa: CFA) -> torch.Tensor:
+    r"""Convert the raw bayer image to RGB version of it and resize width and height by half.
+
+    This is done efficiently by converting each superpixel of bayer image to the corresponding rgb triplet.
+    R and B channels of the raw image are left as are, while two G channels of raw image are averaged to obtain the
+    output G channel.
+
+    We are assuming a CFA with 2 green, 1 red, 1 blue.
+    The image data is assumed to be in the range of (0, 1). Image H/W is assumed to be evenly divisible by 2
+    for simplicity reasons.
+
+    Args:
+        image: raw image to be converted to RGB and downscaled with shape :math:`(*,1,H,W)`.
+        cfa: The configuration of the color filter.
+    Returns:
+        downscaled RGB version of the image with shape :math:`(*,3,\frac{H}{2},\frac{W}{2})`.
+
+    Example:
+        >>> rawinput = torch.randn(2, 1, 4, 6)
+        >>> rgb = raw_to_rgb_2x2_downscaled(rawinput, CFA.RG) # 2x3x2x3
+    """
+    if not isinstance(image, torch.Tensor):
```

Comment:
can you use `KORNIA CHECK` here for type, shape, and checks?  -- https://github.com/kornia/kornia/blob/master/kornia/core/check.py

Reply:
Done